unit_tests/test_aggregator.py

```diff
@@ -0,0 +1,415 @@
+        area_name = "TestArea"
+
+        patch("gsy_e_sdk.aggregator.get_uuid_from_area_name_in_tree_dict",
+              return_value="test_area_uuid")
+        assert aggregator_explicit.get_uuid_from_area_name(area_name) is None
+
+
+class TestCalculateGridFee:
+    """Includes methods which test calculate_grid_fee method."""
+
+    @staticmethod
+    def test_calculate_grid_fee__called(aggregator_explicit):
+        with patch("gsy_e_sdk.aggregator.GridFeeCalculation.calculate_grid_fee") as mocked_method:
+            aggregator_explicit.calculate_grid_fee("start_market",
+                                                   "target_market",
+                                                   "current_market_fee")
+            mocked_method.assert_called_with("start_market",
+                                             "target_market",
+                                             "current_market_fee")
+
+    @staticmethod
+    def test_calculate_grid_fee__return(aggregator_explicit):
+        with patch("gsy_e_sdk.aggregator.GridFeeCalculation.calculate_grid_fee",
+                   return_value="test_ok"):
+            ret_val = aggregator_explicit.calculate_grid_fee("start_market",
+                                                             "target_market",
+                                                             "current_market_fee")
+            assert ret_val == "test_ok"
+
+
+class TestExecuteBatchCommands:
+    """Includes methods which test execute_batch_commands method."""
+
+    @staticmethod
+    def test_execute_batch_commands__command_buffer_length_zero(aggregator_explicit):
+        with patch("gsy_e_sdk.aggregator.ClientCommandBuffer.buffer_length",
+                   new_callable=PropertyMock, return_value=0):
+            assert aggregator_explicit.execute_batch_commands() is None
+
+    @staticmethod
+    def test_execute_batch_commands__execute_batch_called(aggregator_explicit):
+        with patch("gsy_e_sdk.aggregator.ClientCommandBuffer.execute_batch") as mocked_method:
+            aggregator_explicit.start_websocket_connection()
+            aggregator_explicit.execute_batch_commands()
+            mocked_method.assert_called_once()
+
+    @staticmethod
+    def test_execute_batch_commands__validate_all_uuids_in_list(aggregator_explicit):
+        with patch("gsy_e_sdk.aggregator.Aggregator"
+                   + "._all_uuids_in_selected_device_uuid_list") as mocked_method:
+            aggregator_explicit.start_websocket_connection()
+            aggregator_explicit.execute_batch_commands()
+            mocked_method.assert_called()
+
+    @staticmethod
+    def test_execute_batch_commands__post_request_called_with_args(aggregator_explicit):
+        args_ = (f"{'/aggregator_prefix/'}batch-commands",
+                 {"aggregator_uuid": test_agg["uuid"], "batch_commands": test_batch_command_dict}
+                 )
+
+        with patch("gsy_e_sdk.aggregator.Aggregator._post_request",
```

Comment:
Nit: no need to do anything here, but I'd like to talk about this.
As you can see, here we "need" to test `_post_request()` (as you're doing). That's because this is an "outgoing command" that will have side-effects somewhere else, and we care about those side-effects.
The strange thing here is that `_post_request` is a "private" method so, in theory, we shouldn't test it. However, I think that the issue here is in the implementation of the class itself (not the test): IMO `_post_request` should actually be a public method (not private) of `RestCommunicationMixin`.

Reply:
I think this is actually my fault in the terminology I've used. I've used "test method X" in two ways that you also described:
1. write a specific test for `method_X` (its implementation)
2. set expectations (e.g. using mocks) regarding a call made to `method_X` by another method that we are testing now.

So yes: I agree that the method `post_request()` should have its own unit tests somewhere (this is point (1) above).
Regarding (2): that's where I say that _in general_ (not always) we shouldn't make assertions about calls to private methods.